--- origsrc/libproxy-0.2.3/src/bindings/python/Makefile.am	2008-06-05 15:56:16.000000000 -0500
+++ src/libproxy-0.2.3/src/bindings/python/Makefile.am	2009-02-03 19:58:57.613125000 -0600
@@ -1 +1 @@
-dist_python_DATA = $(top_srcdir)/src/bindings/python/libproxy.py
+dist_python_PYTHON = $(top_srcdir)/src/bindings/python/libproxy.py
--- origsrc/libproxy-0.2.3/src/bindings/python/libproxy.py	2008-06-05 15:56:16.000000000 -0500
+++ src/libproxy-0.2.3/src/bindings/python/libproxy.py	2009-02-03 20:39:10.956875000 -0600
@@ -27,6 +27,8 @@ import sys
 # Load libpython (for a cross platform 'free()')
 _names = ("python%d.%d" % sys.version_info[:2], "python%d%d" % sys.version_info[:2])
 _libpython = None
+if sys.platform == 'cygwin':
+	_libpython = ctypes.cdll.LoadLibrary("libpython%d.%d.dll" % sys.version_info[:2])
 for _name in _names:
     if ctypes.util.find_library(_name):
         _libpython = ctypes.cdll.LoadLibrary(ctypes.util.find_library(_name))
