--- origsrc/libproxy-0.2.3/src/lib/Makefile.am	2008-06-05 15:56:15.000000000 -0500
+++ src/libproxy-0.2.3/src/lib/Makefile.am	2009-02-03 19:53:55.706875000 -0600
@@ -3,7 +3,7 @@ lib_LTLIBRARIES     = libproxy.la
 						misc.h url.h pac.h dhcp.h dns.h slp.h wpad.h proxy_factory.h proxy.h config_file.h
 libproxy_la_CFLAGS  = -Wall
 libproxy_la_LDFLAGS = -no-undefined
-libproxy_la_LIBADD  = -lm
+libproxy_la_LIBADD  = -lm -lgetaddrinfo
 
 include_HEADERS     = proxy.h
 
--- origsrc/libproxy-0.2.3/src/lib/proxy_factory.c	2008-06-05 15:56:15.000000000 -0500
+++ src/libproxy-0.2.3/src/lib/proxy_factory.c	2009-02-03 19:55:10.253750000 -0600
@@ -29,6 +29,13 @@
 #include <arpa/inet.h>
 #include <pthread.h>
 
+#ifdef __CYGWIN__
+#include <dummyin6.h>
+#endif
+
+#ifndef RTLD_LOCAL
+#define RTLD_LOCAL 0
+#endif
 
 #include "misc.h"
 #include "proxy_factory.h"
--- origsrc/libproxy-0.2.3/src/lib/url.c	2008-06-05 15:56:15.000000000 -0500
+++ src/libproxy-0.2.3/src/lib/url.c	2009-02-03 19:54:23.253750000 -0600
@@ -26,6 +26,10 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
+#ifdef __CYGWIN__
+#include <dummyin6.h>
+#include <getaddrinfo.h>
+#endif
 
 #include "misc.h"
 #include "url.h"
--- origsrc/libproxy-0.2.3/src/plugins/mozjs.c	2008-06-05 15:56:14.000000000 -0500
+++ src/libproxy-0.2.3/src/plugins/mozjs.c	2009-02-03 19:53:43.535000000 -0600
@@ -25,6 +25,10 @@
 #include <arpa/inet.h>
 #define __USE_BSD
 #include <unistd.h>
+#ifdef __CYGWIN__
+#include <dummyin6.h>
+#include <getaddrinfo.h>
+#endif
 
 #include <misc.h>
 #include <proxy_factory.h>
--- origsrc/libproxy-0.2.3/src/plugins/webkit.c	2008-06-05 15:56:14.000000000 -0500
+++ src/libproxy-0.2.3/src/plugins/webkit.c	2009-02-03 19:55:32.894375000 -0600
@@ -25,6 +25,10 @@
 #include <arpa/inet.h>
 #define __USE_BSD
 #include <unistd.h>
+#ifdef __CYGWIN__
+#include <dummyin6.h>
+#include <getaddrinfo.h>
+#endif
 
 #include <misc.h>
 #include <proxy_factory.h>
