--- origsrc/libproxy-0.4.0/CMakeLists.txt	2010-02-25 14:29:54.000000000 -0600
+++ src/libproxy-0.4.0/CMakeLists.txt	2010-05-06 23:03:34.586143000 -0500
@@ -38,7 +38,7 @@ if(${WIN32})
 else()
   set(rlibdir lib${LIB_SUFFIX}/)
   set(rmoduledir  ${rlibdir}/${CMAKE_PROJECT_NAME}/${CMAKE_PROJECT_VERSION}/modules/)
-  set(rlibexecdir libexec/)
+  set(rlibexecdir ${rlibdir}/${CMAKE_PROJECT_NAME}/${CMAKE_PROJECT_VERSION})
   set(rbindir     bin/)
   set(rincludedir include/)
 endif()
