--- origsrc/libproxy-0.4.14/libproxy/CMakeLists.txt	2017-01-09 08:22:22.000000000 -0600
+++ src/libproxy-0.4.14/libproxy/CMakeLists.txt	2017-03-14 10:49:33.324098400 -0500
@@ -6,7 +6,7 @@ set_project_version(0 4 14)
 if (WIN32)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS=1)
 else(WIN32)
-  add_definitions(-D_POSIX_C_SOURCE=200112L)
+  add_definitions(-D_POSIX_C_SOURCE=200112L -D_XOPEN_SOURCE=600)
   set(CMAKE_CXX_FLAGS "-fvisibility=hidden ${CMAKE_CXX_FLAGS}")
 endif(WIN32)
 
