--- origsrc/libproxy-0.4.0/libproxy/modules/config_gnome.cpp	2010-02-25 14:29:54.000000000 -0600
+++ src/libproxy-0.4.0/libproxy/modules/config_gnome.cpp	2010-05-06 22:56:18.933225100 -0500
@@ -114,7 +114,7 @@ public:
 			throw runtime_error("Unable to open gconf helper!");
 
 		// Set the read pipe to non-blocking
-		if (fcntl(fileno(this->read), F_SETFL, FNONBLOCK) == -1) {
+		if (fcntl(fileno(this->read), F_SETFL, O_NONBLOCK) == -1) {
 			fclose(this->read);
 			fclose(this->write);
 			kill(this->pid, SIGTERM);
