--- origsrc/libproxy-0.4.0/bindings/CMakeLists.txt	2010-02-25 14:29:54.000000000 -0600
+++ src/libproxy-0.4.0/bindings/CMakeLists.txt	2010-05-07 00:22:59.530682200 -0500
@@ -1,12 +1,18 @@
 find_package(PythonInterp)
 
 if(PYTHONINTERP_FOUND)
+  add_custom_command(OUTPUT libproxy.pyc libproxy.pyo
+                     COMMAND ${PYTHON_EXECUTABLE} -m py_compile libproxy.py
+                     COMMAND ${PYTHON_EXECUTABLE} -OO -m py_compile libproxy.py
+                     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                     COMMENT "Byte-compiling python bindings")
+  add_custom_target(python_bytecode ALL DEPENDS libproxy.pyc libproxy.pyo)
   execute_process(COMMAND
                   ${PYTHON_EXECUTABLE}
                   -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(plat_specific=1)"
                   OUTPUT_VARIABLE pysitepkgdir
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
-  install(FILES libproxy.py DESTINATION ${pysitepkgdir})
+  install(FILES libproxy.py libproxy.pyc libproxy.pyo DESTINATION ${pysitepkgdir})
 endif()
 
 #TODO c-sharp bindings
