--- origsrc/libproxy-0.4.0/libmodman/CMakeLists.txt	2010-02-25 14:29:54.000000000 -0600
+++ src/libproxy-0.4.0/libmodman/CMakeLists.txt	2010-05-06 23:08:03.179505700 -0500
@@ -3,8 +3,9 @@ add_library(libmodman SHARED module.hpp
 if(NOT WIN32)
   target_link_libraries(libmodman dl)
 endif()
-set_target_properties(libmodman PROPERTIES PREFIX "" VERSION 0 SOVERSION 0.0.0)
-install(TARGETS libmodman DESTINATION ${rlibdir})
+set_target_properties(libmodman PROPERTIES OUTPUT_NAME "modman" VERSION 0.0.0 SOVERSION 0)
+install(TARGETS libmodman LIBRARY DESTINATION ${rlibdir}
+                          RUNTIME DESTINATION bin)
 
 ### Tests
 # Functions
--- origsrc/libproxy-0.4.0/libproxy/CMakeLists.txt	2010-02-25 14:29:54.000000000 -0600
+++ src/libproxy-0.4.0/libproxy/CMakeLists.txt	2010-05-16 14:50:20.604734300 -0500
@@ -139,7 +139,7 @@ message("")
 # PkgConfig file
 if(NOT ${WIN32})
   configure_file(libproxy-1.0.pc.in libproxy-1.0.pc @ONLY)
-  install(FILES libproxy-1.0.pc DESTINATION ${libdir}/pkgconfig)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libproxy-1.0.pc DESTINATION ${libdir}/pkgconfig)
 endif()
 
 # GNOME (gconf) helper
@@ -163,6 +163,7 @@ if(WIN32)
 endif()
 
 set_property(SOURCE proxy.cpp PROPERTY COMPILE_DEFINITIONS MODULEDIR="${moduledir}";BUILTIN_MODULES=${BUILTIN_MODULES})
-set_target_properties(libproxy PROPERTIES PREFIX "" VERSION 1.0.0 SOVERSION 1)
-install(TARGETS libproxy DESTINATION ${rlibdir})
+set_target_properties(libproxy PROPERTIES OUTPUT_NAME "proxy" VERSION 1.0.0 SOVERSION 1)
+install(TARGETS libproxy LIBRARY ARCHIVE DESTINATION ${rlibdir}
+                         RUNTIME DESTINATION bin)
 install(FILES   proxy.h  DESTINATION ${rincludedir})
